src/provider/provider_os_memory.c

```diff
@@ -437,32 +517,34 @@ static umf_result_t os_alloc(void *provider, size_t size, size_t alignment,
     }
 
     errno = 0;
-    if (hwloc_bitmap_iszero(os_provider->nodeset)) {
-        // Hwloc_set_area_membind fails if empty nodeset is passed so if no node is specified,
-        // just pass all available nodes. For modes where no node is needed, they will be
-        // ignored anyway.
-        hwloc_const_nodeset_t complete_nodeset =
-            hwloc_topology_get_complete_nodeset(os_provider->topo);
-        ret = hwloc_set_area_membind(os_provider->topo, addr, size,
-                                     complete_nodeset, os_provider->numa_policy,
-                                     os_provider->numa_flags);
-    } else {
+    unsigned membind = get_membind(os_provider, ALIGN_UP(size, page_size));
+    size_t bind_size = os_provider->nodeset_len == 1
```

Comment:
No - if part_size is set we use our algorithm instead of os one - if user wants OS one, then they do not set part size(or set it to 0), and then we rely on OS 
Rationale:
- I want to be consistent 
- There might be a (transparent)huge pages.
- OS interleave works in the different way then our bind, so i do not want special value, to change algorithm behavior.